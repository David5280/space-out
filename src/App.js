import React from 'react';
import { getSearchData } from './Utilz/apiCalls';
import Header from './Header/Header';
import Aside from './Aside/Aside';
import './SASS/Index.scss';

class App extends React.Component {
  componentDidMount() {
    getSearchData()
    .then(data => console.log('a', data))
    .catch(error => error.message)
  }
  render() {
    return (
      <div className="App">
        <Header />
        <Aside />
      </div>
    )
  }
}

export default App;
