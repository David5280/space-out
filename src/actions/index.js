export const loadPlanets = (planets) => ({
  type: 'LOAD_PLANETS',
  planets
});

export const loadComplete = () => ({
  type: 'LOAD_COMPLETE'
});

export const addUser = (user) => ({
  type: 'ADD_USER',
  user
});