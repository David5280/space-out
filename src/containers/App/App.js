import React from 'react';
import Header from '../../components/Header/Header';
import MainDisplay from '../MainDisplay/MainDisplay';
import LoadingDisplay from '../../components/LoadingDisplay/LoadingDisplay';
import PropTypes from 'prop-types';
import { loadPlanets, loadComplete } from '../../actions';
import { getData } from '../../Utilz/apiCalls';
import { connect } from 'react-redux';
import '../../SASS/Index.scss';

export class App extends React.Component {
  componentDidMount() {
    this.retrieveData()
  }

  retrieveData = () => {
    const { loadPlanets, loadComplete } = this.props;
    getData()
    .then(data => loadPlanets(data))
    .then(() => loadComplete())
    .catch(error => error.message)
  }

  render() {
    return (
      <div className="App">
        <Header planets={this.props.planets}/>
        {this.props.loading && <LoadingDisplay />}
        {this.props.planets && <MainDisplay planets={this.props.planets} />}
      </div>
    )
  }
};

export const mapStateToProps = (state) => ({
  planets:  state.planets,
  loading: state.loading
});

export const mapDispatchToProps = (dispatch) => ({
  loadPlanets: (planets) => dispatch(loadPlanets(planets)),
  loadComplete: () => dispatch(loadComplete())
});

App.propTypes = {
  planets: PropTypes.array,
  loading: PropTypes.bool,
  loadPlanets: PropTypes.func,
  loadComplete: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
