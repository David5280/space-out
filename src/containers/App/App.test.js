import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { loadPlanets, loadComplete } from '../../actions'; 

jest.mock('../../utilz/apiCalls', () => ({
  getData: jest.fn().mockImplementation(() => {
    return [{ title: 'earth' }, { title: 'mars' }];
  })
}));

describe('App', () => {
  describe('App Component', () => {
    let wrapper;
    beforeEach(() => {
      const props = {
        loadPlanets: jest.fn(),
      }
      wrapper = shallow(<App {...props} />, {disableLifecycleMethods: true})
    })
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should call retrieveData when component mounts', () => {
      wrapper.instance().retrieveData = jest.fn()
      wrapper.instance().componentDidMount();
      expect(wrapper.instance().retrieveData).toHaveBeenCalled()
    });
  });
  describe('mapStateToProps', () => {
    it('should return an object with the planets array', () => {
      const mockState = {
        planets: [{ title: 'mars' }]
      }
      const expected = {
        planets: [{ title: 'mars' }]
      }
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });
  describe('mapDispatchToProps', () => {
    it('calls dispatch with planets when loadPlanets is called', () => {
      const mockPlanets = [
        { title: 'jupiter' },
        { title: 'pluto' }
      ];
      const mockDispatch = jest.fn();
      const actionToDispatch = loadPlanets(mockPlanets);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.loadPlanets(mockPlanets);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
    it('calls dispatch when loadComplete is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = loadComplete();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.loadComplete();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
