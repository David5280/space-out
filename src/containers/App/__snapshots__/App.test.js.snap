// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App Component should match snapshot 1`] = `ShallowWrapper {}`;
